js/client-battle.js

```diff
@@ -1083,6 +1112,202 @@
+				if (pokemon.ability === 'Refrigerate') moveType = 'Ice';
+			}
+			return moveType;
+		},
+		// Gets the proper current base power for moves which have a variable base power.
+		// Takes into account the target for some moves.
+		// If it is unsure of the actual base power, it gives an estimate.
+		getMoveBasePower: function(move, pokemon, target) {
+			var basePower = move.basePower;
+			var thereIsWeather = (this.battle.weather in {'sunnyday': 1, 'desolateland': 1, 'raindance': 1, 'primordialsea': 1, 'sandstorm': 1, 'hail':1});
+			if (move.id === 'acrobatics') {
+				if (!pokemon.item) basePower = (basePower * 2) + ' (Boosted by lack of item)' ;
+			}
+			if (move.id === 'crushgrip' || move.id === 'wringout') {
+				basePower = Math.floor(Math.floor((120 * (100 * Math.floor(target.hp * 4096 / target.maxhp)) + 2048 - 1) / 4096) / 100) || 1;
+				basePower += ' (Approximation)';
+			}
+			if (move.id === 'eruption' || move.id === 'waterspout') {
+				basePower = Math.floor(150 * pokemon.hp / pokemon.maxhp) || 1;
+			}
+			if (move.id === 'flail' || move.id === 'reversal') {
+				if (this.battle.gen > 4) {
+					var multiplier = 48;
+					var ratios = [2, 5, 10, 17, 33];
+				} else {
+					var multiplier = 64;
+					var ratios = [2, 6, 13, 22, 43];
+				}
+				var ratio = pokemon.hp * multiplier / pokemon.maxhp;
+				if (ratio < ratios[0]) basePower = 200;
+				else if (ratio < ratios[1]) basePower = 150;
+				else if (ratio < ratios[2]) basePower = 100;
+				else if (ratio < ratios[3]) basePower = 80;
+				else if (ratio < ratios[4]) basePower = 40;
+				else basePower = 20;
+			}
+			if (move.id === 'hex') {
+				if (target.status) basePower = '130 (Boosted by status)';
+				basePower = 65;
+			}
+			if (move.id === 'punishment') {
+				var boosts = Object.keys(target.boosts);
+				var multiply = 0;
+				for (var i = 0; i < boosts.length; i++) {
+					if (target.boosts[boosts[i]] && target.boosts[boosts[i]] > 0) multiply++;
+				}
+				basePower = 60 + 20 * multiply;
+			}
+			if (move.id === 'smellingsalts') {
+				if (target.status === 'par') basePower = (basePower * 2) + ' (Boosted by status)' ;
+			}
+			if (move.id === 'storedpower') {
+				var boosts = Object.keys(pokemon.boosts);
+				var multiply = 0;
+				for (var i = 0; i < boosts.length; i++) {
+					if (pokemon.boosts[boosts[i]] && pokemon.boosts[boosts[i]] > 0) multiply++;
+				}
+				basePower = 20 + 20 * multiply;
+			}
+			if (move.id === 'trumpcard') {
+				basePower = 40;
+				if (move.pp === 1) basePower = 200;
+				else if (move.pp === 2) basePower = 80;
+				else if (move.pp === 3) basePower = 60;
+				else if (move.pp === 4) basePower = 50;
+			}
+			if (move.id === 'venoshock') {
+				if (target.status === 'psn' || target.status === 'tox') basePower = (basePower * 2) + ' (Boosted by status)' ;
+			}
+			if (move.id === 'wakeupslap') {
+				if (target.status === 'slp') basePower = (basePower * 2) + ' (Boosted by status)' ;
```

Comment:
I thought that because those are moves that are boosted specifically by a status only if you use those moves, as opposed to a general boost to damage due to status (Guts), while weakness / resistance being something to account for separatedly. Want me to add an extra next to STAB about resistances?